src/radical/pilot/pmgr/launching/base.py

```diff
@@ -286,6 +286,13 @@ def work(self, pilots):
         for pilot in pilots:
             resource = pilot['description']['resource']
             schema   = pilot['description']['access_schema']
+            if schema == 'interactive' and 'ornl' in resource:
+                smt_cmd = r""
+                smt_cmd += r"expr $(jsrun -n1 -bpacked:1 /bin/bash -c 'echo $OMP_PLACES') :"
+                smt_cmd += r" '.*\(.\)}' | tr 0 1;"
+                smt = ru.sh_callout(smt_cmd, shell=True)[0].strip()
+                os.environ["RADICAL_SMT"] = smt
```

Comment:
Sorry for late response!  I would prefer if we do not need to rely on the user to provide the export.  `RADICAL_SMT` is great if it exists, but otherwise I agree we should either inspect the system for the setting (as we discussed in some place) or fall back to the configs.  A prominent log message (`logger.info('used SMT setting: %d' % smt`) would be useful.

Reply:
Andre, but such code with `jsrun` command we shouldn't have in PMGRLaunching, right? It should get such info from the config (that what I would propose to have -> `cfg.pmgr_pre_launch`) and then execute it, thus it will set necessary env variable

Reply:
@mtitov and @andre-merzky discussed this again.  We decided that both on pmgr and on agent side we check `RADICAL_SMT`, and if not set fall back to the config value.  It is the user's responsibility to ensure either.

This also implies that `RADICAL_SMT` should be consistently set on the SAGA level, including in the SHELL adaptor.

Reply:
@mtitov : I pushed a commit for the above, can you please check if that matches our discussion?
This should actually work w/o changes to the SAGA adaptors as `RADICAL_SMT` ends up in the pilot environment anyway and the bootstrapper picks it up from there.  But supporting in in the shell adaptor is cleaner and makes sense either way...

Reply:
added couple comments on that updates (mostly related to the provided example for Summit, and it is not the case for Frontera)